hapi-fees/src/main/java/com/hedera/services/usage/crypto/CryptoOpsUsage.java

```diff
@@ -178,4 +178,32 @@ public void cryptoCreateUsage(final SigUsage sigUsage,
 		accumulator.addRbs(maxAutomaticTokenAssociations * lifeTime * CREATE_SLOT_MULTIPLIER);
 		accumulator.addNetworkRbs(BASIC_ENTITY_ID_SIZE * USAGE_PROPERTIES.legacyReceiptStorageSecs());
 	}
+
+	public void cryptoApproveAllowanceUsage(final SigUsage sigUsage,
+			final BaseTransactionMeta baseMeta,
+			final CryptoApproveAllowanceMeta cryptoApproveMeta,
+			final ExtantCryptoContext ctx,
+			final UsageAccumulator accumulator) {
+		accumulator.resetForTransaction(baseMeta, sigUsage);
+		accumulator.addBpt(cryptoApproveMeta.getMsgBytesUsed());
+
+		var newVariableSerialBytes = 0;
+		newVariableSerialBytes += cryptoApproveMeta.getAggregatedNftAllowancesWithSerials() * LONG_SIZE;
+		accumulator.addRbs(newVariableSerialBytes);
+
+		final long sharedFixedBytes = CRYPTO_ENTITY_SIZES.fixedBytesInAccountRepr();
+
+		final long newLifetime = ESTIMATOR_UTILS.relativeLifetime(
```

Comment:
Since a `cryptoApproveAllowance` can't change the expiry of an account (the way a `cryptoUpdate` can), we only need one lifetime calculation. 👍

Reply:
Fixed . Thanks !